ui/core/src/utils/time-series-data.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { AbsoluteTimeRange, TimeScale, TimeSeries, TimeSeriesData } from '../model';
+import { gcd } from './mathjs';
+
+export const MIN_STEP_INTERVAL_MS = 10;
+
```

Comment:
This code is moved from `panels-plugin/src/plugins/time-series-chart/utils/data-transform.ts`.  I made a few tweaks to the jsdoc to clarify some expectations, but the code itself is unchanged.